R/pkg/inst/worker/daemon.R

```diff
@@ -47,8 +55,6 @@ while (TRUE) {
       close(inputCon)
       Sys.setenv(SPARKR_WORKER_PORT = port)
       try(source(script))
-      # Set SIGUSR1 so that child can exit
-      tools::pskill(Sys.getpid(), tools::SIGUSR1)
       parallel:::mcexit(0L)
```

Comment:
`mcexit` alone does not terminate the child.


With terminal A:

```R
R
```

With terminal B:

```bash
watch -n 0.01 "ps -fe | grep /bin/exec/R"
```

with terminal A:

```R
for(i in 0:100) {
  p <- parallel:::mcfork()
  if (inherits(p, "masterProcess")) {
    parallel:::mcexit(0L)
  }
}
```